node/src/database/rocksdb.rs

```diff
@@ -7,130 +7,350 @@
 use super::{Registry, Tx, DB};
 use anyhow::Result;
 
+use dusk_consensus::commons::Block;
+use dusk_consensus::messages::Serializable;
+use rocksdb_lib::{
+    ColumnFamily, DBAccess, DBCommon, DBWithThreadMode, MultiThreaded,
+    OptimisticTransactionDB, OptimisticTransactionOptions, Options,
+    ReadOptions, SnapshotWithThreadMode, WriteOptions,
+};
+use std::{marker::PhantomData, path::Path, sync::Arc};
+
 enum TxType {
     ReadWrite,
     ReadOnly,
 }
 
-pub struct Backend {}
+const CF_LEDGER: &str = "cf_ledger";
+const CF_CANDIDATES: &str = "cf_candidates";
+
+pub struct Backend {
+    rocksdb: Arc<OptimisticTransactionDB>,
+}
 
 impl Backend {
-    fn begin_tx(&mut self, read_only: TxType) -> Transaction {
-        // TODO: rusk/issues/806 This should be addressed with another issue
-        // about integrating RocksDB
-        Transaction {}
+    fn begin_txn(
+        &self,
+        access_type: TxType,
+    ) -> Transaction<'_, OptimisticTransactionDB> {
+        // Create a new RocksDB transaction
+        let write_options = WriteOptions::default();
+        let mut txn_options = OptimisticTransactionOptions::default();
+
+        let inner = self.rocksdb.transaction_opt(&write_options, &txn_options);
+
+        // Borrow column families
+        let ledger_cf = self
+            .rocksdb
+            .cf_handle(CF_LEDGER)
+            .expect("ledger column family must exist");
+
+        let candidates_cf = self
+            .rocksdb
+            .cf_handle(CF_CANDIDATES)
+            .expect("candidates column family must exist");
+
+        let snapshot = self.rocksdb.snapshot();
+
+        Transaction::<'_, OptimisticTransactionDB> {
+            inner,
+            access_type,
+            candidates_cf,
+            ledger_cf,
+            snapshot,
+        }
     }
 }
 
 impl DB for Backend {
-    type T = Transaction;
+    type T<'a> = Transaction<'a, OptimisticTransactionDB>;
```

Comment:
Get compilation error here.
Better to add the compilation of the crate to the generic `make test` otherwise it's not spotted by the CI

```
error[E0658]: generic associated types are unstable
  --> node/src/database/rocksdb.rs:66:5
   |
66 |     type T<'a> = Transaction<'a, OptimisticTransactionDB>;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #44265 <https://github.com/rust-lang/rust/issues/44265> for more information
   = help: add `#![feature(generic_associated_types)]` to the crate attributes to enable
```

Reply:
Good catch! I was using a more recent rust toolchain. I see that GATs are enabled in 1.65 (as per [link]( https://blog.rust-lang.org/2022/11/03/Rust-1.65.0.html#generic-associated-types-gats) so for now I'll just enable the feature.


Meanwhile, another issue #821  will add this crate to Rust CI procedure.